python_modules/dagster/dagster/config/stack.py

```diff
@@ -24,6 +24,12 @@ def for_field(self, field_name: str) -> "EvaluationStack":
     def for_array_index(self, list_index: int) -> "EvaluationStack":
         return EvaluationStack(entries=self.entries + [EvaluationStackListItemEntry(list_index)])
 
+    def for_map_key(self, map_key: object) -> "EvaluationStack":
+        return EvaluationStack(entries=self.entries + [EvaluationStackMapKeyEntry(map_key)])
+
+    def for_map_value(self, map_key: object) -> "EvaluationStack":
+        return EvaluationStack(entries=self.entries + [EvaluationStackMapValueEntry(map_key)])
```

Comment:
should map_key be map_value here? I see it has keys passed in in its callsites, but those are themselves... values, right?

Reply:
I'm not sure I fully understand the comment, but the reason that the map's key is passed into `EvaluationStackMapKeyEntry` and `EvaluationStackMapValueEntry` is that the key is the unique identifier that is printed in error messages and identifies the place in the stack, even if the value is what has the error. So, key `x` with the ValueEntry specifies that you are processing the value for key `x`, key `x` with the KeyEntry specifies that you're processing `x` itself.